libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,657 @@
+ * It is used to display a list features.
+ */
+@Component({
+    selector: 'fdp-list',
+    templateUrl: './list.component.html',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: ListComponent, multi: true }]
+})
+
+
+export class ListComponent extends CollectionBaseInput implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
+
+    /**Enables lazy loadMore of data */
+    @Input()
+    loadMore: boolean;
+
+    /**Enables data load on scroll for true
+     * false: enables data loading on button
+     * click
+     */
+    @Input()
+    loadOnScroll: boolean;
+
+    /**Wait time for new items */
+    @Input()
+    delayTime: number;
+
+    /**Max height for viewport to display scroll */
+    @Input()
+    maxHeight: string;
+
+    /**Items to be loaded at once */
+    @Input()
+    itemSize: number;
+
+    /**Scroll offset percentage to trigger scroll event */
+    @Input()
+    scrollOffsetPercentage: number;
+
+    /** Whether list component has removed borders */
+    @Input()
+    noBorder = false;
+
+    @Input()
+    selection = false;
+
+    /** Whether list component has removed bottom borders */
+    @Input()
+    noSeperator: boolean;
+
+    /** The type of the selection. Types include:
+    *'none'| 'multi' | 'single'|'delete'.
+    * Leave empty for default ().'
+    * Default value is set to ''
+    */
+    @Input()
+    selectionMode: SelectionType = 'none';
+
+    /**ListType 'inactive' | 'active' | 'navigation' | 'detail' */
+    @Input()
+    listType: ListType = 'active';
+
+    /**  An array that holds a list of all selected items**/
+    @Input()
+    selectedItems: BaseListItem[];
+
+    /** define size of items for screen reader */
+    @Input()
+    ariaSetsize: number;
+
+    /** Defines whether items are multiseletable for screen reader */
+    @Input()
+    ariaMultiselectable: boolean;
+
+    /**Title used on button when data loads on button click */
+    @Input()
+    loadTitle: string;
+
+    /**
+   * Todo: Name of the entity for which DataProvider will be loaded.
+   *  You can either pass list of
+   * items or use this entityClass and internally we should be able
+   * to do lookup to some registry
+   * and retrieve the best matching DataProvider that is set on application level
+   *
+   *
+   */
+    @Input()
+    entityClass: string;
+
+
+    /**
+      * Child items of the List.
+      */
+    @ContentChildren(BaseListItem, { descendants: true }) ListItems: QueryList<BaseListItem>;
+
+    @ContentChild(ListItemDef) listItemDef: ListItemDef;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    /**
+     * @hidden
+     * keyManger to handle keybord events
+     */
+    keyManager: FocusKeyManager<BaseListItem>;
```

Comment:
Is this used internally? Please see my comment bellow about naming it with `_` prefix to express this internal API,

I see this hidden, but its public? 

Please go thru the rest of properties

Reply:
KeyManager is used in list.component.html marking it to private/  _ are leading compilation errors, so I made it hidden.

In previous reviews, this concern was raised to align members in order. So after referring angular coding guidelines. I was following the below sequence. 

Member Sequence:

@Input
Public used in a template 
Private
@viewchild
@contentchild
@output
Getter and setter
Public functions
Public functions hidden minded with template
Lifecycle
Private functions

For file import:
angular
external
internal

Please do let me know if this is not correct, i will update accordingly.